test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs

```diff
@@ -1039,6 +1041,47 @@ public void SdkPopulatesSamplingParamsCorrectlyForLegacyActivityWithInProcParent
         activity.Stop();
     }
 
+    [Theory]
+    [InlineData(null, null, "ParentBased{AlwaysOnSampler}")]
+    [InlineData("always_on", null, "AlwaysOnSampler")]
+    [InlineData("always_off", null, "AlwaysOffSampler")]
+    [InlineData("always_OFF", null, "AlwaysOffSampler")]
+    [InlineData("traceidratio", "0.5", "TraceIdRatioBasedSampler{0.500000}")]
+    [InlineData("traceidratio", "not_a_double", "TraceIdRatioBasedSampler{1.000000}")]
+    [InlineData("parentbased_always_on", null, "ParentBased{AlwaysOnSampler}")]
+    [InlineData("parentbased_always_off", null, "ParentBased{AlwaysOffSampler}")]
+    [InlineData("parentbased_traceidratio", "0.111", "ParentBased{TraceIdRatioBasedSampler{0.111000}}")]
+    [InlineData("ParentBased_TraceIdRatio", "0.000001", "ParentBased{TraceIdRatioBasedSampler{0.000001}}")]
+    public void TestSamplerSetFromConfiguration(string configValue, string argValue, string samplerDescription)
+    {
+        var configBuilder = new ConfigurationBuilder();
+
+        if (!string.IsNullOrEmpty(configValue))
+        {
+            configBuilder.AddInMemoryCollection(new Dictionary<string, string>
+            {
+                [TracerProviderSdk.TracesSamplerConfigKey] = configValue,
+            });
+        }
+
+        if (!string.IsNullOrEmpty(argValue))
+        {
+            configBuilder.AddInMemoryCollection(new Dictionary<string, string>
+            {
+                [TracerProviderSdk.TracesSamplerArgConfigKey] = argValue,
+            });
+        }
+
+        var builder = Sdk.CreateTracerProviderBuilder();
+        builder.ConfigureServices(s => s.AddSingleton<IConfiguration>(configBuilder.Build()));
```

Comment:
@matt-hensley Might be a good idea to add a permutation to this test which calls `builder.SetSampler` and then verifies the config is ignored for that case?